### PR TITLE
[Model] Implement Grace-POT

### PR DESCRIPTION
**Pull request by GuoFeng97**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by GuoFeng97** [comment]
10.19 gf grace-pot

File: examples/grace-pot/GRACE_POT_trainer.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+import tensorlayerx as tlx
+from tensorlayerx.model import TrainOneStep, WithLoss
+from tensorlayerx.dataflow import random_split
+
+from gammagl.utils import set_device
+from gammagl.layers.conv import GCNConv
+from gammagl.datasets import Planetoid, Coauthor, Amazon
+import gammagl.transforms as T
+
+from grace_pot import Encoder, Model
```
Comment (dddg617) [inline examples/grace-pot/GRACE_POT_trainer.py:19]: There is an error here, this should be `gammagl.models.grace-pot`.

File: examples/grace-pot/eval_gracepot.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
+from typing import Optional
+import numpy as np
+import tensorlayerx as tlx
+from tensorlayerx.optimizers import Adam
+import torch
+from model_ggl import LogReg
```
Comment (dddg617) [inline examples/grace-pot/eval_gracepot.py:6]: There is also an error here.

Comment (dddg617) [reply examples/grace-pot/eval_gracepot.py:6]: I recommend the LogReg in `grace_pot` to be moved here.

File: gammagl/models/grace_pot.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+import scipy.sparse as sp
+import pickle
+from scipy.sparse import coo_matrix
+import os.path as osp
+from time import perf_counter as t
+from gammagl.mpops import unsorted_segment_sum
+import tensorflow
+
+def BCEWithLogitsLoss(output, target):
+    return torch.nn.BCEWithLogitsLoss()(output, target)
```
Comment (dddg617) [inline gammagl/models/grace_pot.py:18]: I think this function is of no use.

File: gammagl/models/grace_pot.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+        row, col = tlx.convert_to_numpy(edge_index_ptb_sl)
+        An_ptb = coo_matrix((An_ptb.cpu(),(row, col)), shape=(A.shape[0],A.shape[0])).toarray()
+        An_ptb=tlx.gather(tlx.convert_to_tensor(An_ptb),tlx.convert_to_tensor(node_list),0)
+        An_ptb=tlx.gather(An_ptb,tlx.convert_to_tensor(node_list),1)
+        An_ptb=tlx.convert_to_tensor(An_ptb)
+        H_tilde = mm(An_ptb, XW_tilde) + b_tilde_1.reshape(-1,1)
+        pot_score = mm(An_ptb, H_tilde) + b_tilde_2.reshape(-1,1)
+        pot_score = tlx.squeeze(pot_score,axis=1)
+        target = tlx.zeros(tlx.get_tensor_shape(pot_score)) + 1
+        pot_loss = BCEWithLogitsLoss(pot_score, target)
```
Comment (dddg617) [inline gammagl/models/grace_pot.py:201]: You may use `tlx.losses.sigmoid_cross_entropy(pot_score, target)`.

File: gammagl/models/grace_pot.py [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+        self.encoder: Encoder = encoder
+        self.tau: float = tau
+        self.dataset = dataset
+
+        self.fc1 = tlx.nn.Linear(in_features=num_hidden, out_features=num_proj_hidden)
+        self.fc2 = tlx.nn.Linear(in_features=num_proj_hidden, out_features=num_hidden)
+    def forward(self, x, edge_index):
+        return self.encoder(x, edge_index)
+
+    def projection(self, z: tlx.convert_to_tensor)->tlx.convert_to_tensor:
```
Comment (dddg617) [inline gammagl/models/grace_pot.py:69]: What do you mean? The type of z is `tlx.convert_to_tensor`? Remember to correct the other similar error.

**Comment by GuoFeng97** [comment]
10.19-19:34-version

**Pull request merged**